File: backend/src/main/java/com/hyunprise/backend/domain/coupon/vo/Coupon.java
package com.hyunprise.backend.domain.coupon.vo;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;


@Data
@AllArgsConstructor
@NoArgsConstructor
public class Coupon {
    private String couponUUID;
    private String sellerUUID;
    private String couponName;
    private String couponDescription;
    private Integer discountType;
    private Integer discountAmount;
    private Timestamp createdDate;
    private Integer discountLimit;
    private Integer minimumPurchase;
    private String retailerLocation;
    private String termsAndConditions;
    private String usageInstruction;
    private Integer expirationDays;
}

File: android/app/src/main/java/com/hyunprise/android/api/coupon/vo/Coupon.kt
package com.hyunprise.android.api.coupon.vo

import java.sql.Timestamp

data class Coupon(
    val couponUUID: String? = null,
    var sellerUUID: String? = null,
    var couponName: String? = null,
    var couponDescription: String? = null,
    var discountType: Int? = null,
    var discountAmount: Int? = null,
    val creationDate: Timestamp? = null,
    var discountLimit: Int? = null,
    var minimumPurchase: Int? = null,
    val retailerLocation: String? = null,
    val termsAndConditions: String? = null,
    val usageInstruction: String? = null,
    var expirationDays: Int? = null
)
